src/graph/RaytraceNode.cpp

```diff
@@ -69,10 +71,19 @@ void RaytraceNode::schedule(cudaStream_t stream)
 
 void RaytraceNode::setFields(const std::set<rgl_field_t>& fields)
 {
-	this->fields = std::move(fields);
+	auto keyViewer = std::views::keys(fieldData);
+	std::set<rgl_field_t> oldKeys { keyViewer.begin(), keyViewer.end() };
+
 	for (auto&& field : fields) {
 		if (!fieldData.contains(field)) {
 			fieldData.insert({field, VArray::create(field)});
 		}
+		if (oldKeys.contains(field)) {
+			oldKeys.erase(field);
+		}
+	}
+
+	for (auto&& keyToErase : oldKeys) {
+		fieldData.erase(keyToErase);
```

Comment:
1. OK, c++20 is very helpful.
2. Rare operation, but it could cause annoying lag when changing graphs in runtime. Common case: developing simulation in game engine.